### PR TITLE
May Release

### PR DESCRIPTION
**Pull request by kevinkowa**

### CODE REVIEW

File: compare-comply/src/main/java/com/ibm/watson/compare_comply/v1/CompareComply.java [hunk 1]
```diff
@@ -55,20 +55,19 @@
...
+ * [https://ibm.biz/contact-wdc-premium](https://ibm.biz/contact-wdc-premium). {: deprecated}
+ *
+ * <p>Compare and Comply analyzes governing documents to provide details about critical aspects of
+ * the documents.
  *
  * @version v1
  * @see <a href="https://cloud.ibm.com/docs/compare-comply?topic=compare-comply-about">Compare
  *     Comply</a>
  */
-
```
Comment (apaparazzi0329) [inline compare-comply/src/main/java/com/ibm/watson/compare_comply/v1/CompareComply.java:65]: It looks like the deprecation message was accidentally deleted

File: compare-comply/src/main/java/com/ibm/watson/compare_comply/v1/CompareComply.java [hunk 2]
```diff
@@ -130,11 +129,6 @@ public CompareComply(String version, String serviceName, Authenticator authentic
     setServiceUrl(DEFAULT_SERVICE_URL);
     setVersion(version);
     this.configureService(serviceName);
-
```
Comment (apaparazzi0329) [inline compare-comply/src/main/java/com/ibm/watson/compare_comply/v1/CompareComply.java:133]: Here as well

File: discovery/src/main/java/com/ibm/watson/discovery/v1/Discovery.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright IBM Corp. 2019, 2020.
```
Comment (apaparazzi0329) [inline discovery/src/main/java/com/ibm/watson/discovery/v1/Discovery.java:2]: This file only has copyright changes so it should have been linted by the fix-copyrights script

File: language-translator/src/main/java/com/ibm/watson/language_translator/v3/LanguageTranslator.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright IBM Corp. 2019, 2020.
```
Comment (apaparazzi0329) [inline language-translator/src/main/java/com/ibm/watson/language_translator/v3/LanguageTranslator.java:2]: Only copyright changes

File: natural-language-classifier/src/main/java/com/ibm/watson/natural_language_classifier/v1/NaturalLanguageClassifier.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright IBM Corp. 2019, 2020.
```
Comment (apaparazzi0329) [inline natural-language-classifier/src/main/java/com/ibm/watson/natural_language_classifier/v1/NaturalLanguageClassifier.java:2]: Only copyright changes

File: personality-insights/src/main/java/com/ibm/watson/personality_insights/v3/PersonalityInsights.java [hunk 6]
```diff
@@ -64,14 +65,6 @@
  * @version v3
  * @see <a href="https://cloud.ibm.com/docs/personality-insights">Personality Insights</a>
  */
-
```
Comment (apaparazzi0329) [inline personality-insights/src/main/java/com/ibm/watson/personality_insights/v3/PersonalityInsights.java:67]: Deprecation message removed

File: personality-insights/src/main/java/com/ibm/watson/personality_insights/v3/PersonalityInsights.java [hunk 7]
```diff
@@ -133,12 +126,6 @@ public PersonalityInsights(String version, String serviceName, Authenticator aut
     setServiceUrl(DEFAULT_SERVICE_URL);
     setVersion(version);
     this.configureService(serviceName);
-
```
Comment (apaparazzi0329) [inline personality-insights/src/main/java/com/ibm/watson/personality_insights/v3/PersonalityInsights.java:136]: Deprecation message removed

File: tone-analyzer/src/main/java/com/ibm/watson/tone_analyzer/v3/ToneAnalyzer.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright IBM Corp. 2016, 2020.
```
Comment (apaparazzi0329) [inline tone-analyzer/src/main/java/com/ibm/watson/tone_analyzer/v3/ToneAnalyzer.java:2]: Only copyright changes

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v3/VisualRecognition.java [hunk 9]
```diff
@@ -55,13 +55,6 @@
  * @version v3
  * @see <a href="https://cloud.ibm.com/docs/visual-recognition">Visual Recognition</a>
  */
-
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v3/VisualRecognition.java:58]: Deprecation message removed

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v3/VisualRecognition.java [hunk 10]
```diff
@@ -123,10 +116,6 @@ public VisualRecognition(String version, String serviceName, Authenticator authe
     setServiceUrl(DEFAULT_SERVICE_URL);
     setVersion(version);
     this.configureService(serviceName);
-
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v3/VisualRecognition.java:126]: Deprecation message removed

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java [hunk 11]
```diff
@@ -79,13 +79,6 @@
  *     href="https://cloud.ibm.com/docs/visual-recognition?topic=visual-recognition-object-detection-overview">Visual
  *     Recognition</a>
  */
-
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java:82]: Deprecation message removed

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java [hunk 12]
```diff
@@ -147,10 +140,6 @@ public VisualRecognition(String version, String serviceName, Authenticator authe
     setServiceUrl(DEFAULT_SERVICE_URL);
     setVersion(version);
     this.configureService(serviceName);
-
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java:150]: Deprecation message removed

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java [hunk 13]
```diff
@@ -201,12 +190,9 @@ public void setVersion(final String version) {
     builder.query("version", String.valueOf(this.version));
     MultipartBody.Builder multipartBuilder = new MultipartBody.Builder();
     multipartBuilder.setType(MultipartBody.FORM);
-    for (String item : analyzeOptions.collectionIds()) {
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/VisualRecognition.java:204]: Make sure this is supposed to be removed and not a purposeful hand edit being removed

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptions.java [hunk 14]
```diff
@@ -25,10 +25,12 @@
   public static class Builder {
     private String name;
     private String description;
+    private TrainingStatus trainingStatus;
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptions.java:28]: The changes in this file need to be removed. TrainingStatus should not be a property here

File: visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/model/UpdateCollectionOptions.java [hunk 15]
```diff
@@ -27,11 +27,13 @@
     private String collectionId;
     private String name;
     private String description;
+    private TrainingStatus trainingStatus;
```
Comment (apaparazzi0329) [inline visual-recognition/src/main/java/com/ibm/watson/visual_recognition/v4/model/UpdateCollectionOptions.java:30]: Same issue here. TrainingStatus should not be here

File: visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/VisualRecognitionTest.java [hunk 16]
```diff
@@ -176,9 +179,28 @@ public void testCreateCollectionWOptions() throws Throwable {
 
     constructClientService();
 
+    // Construct an instance of the ObjectTrainingStatus model
```
Comment (apaparazzi0329) [inline visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/VisualRecognitionTest.java:182]: Changes in this file involving TrainingStatus need to be removed

File: visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptionsTest.java [hunk 17]
```diff
@@ -51,9 +51,13 @@ public void testCreateCollectionOptions() throws Throwable {
     assertEquals(trainingStatusModel.objects(), objectTrainingStatusModel);
 
     CreateCollectionOptions createCollectionOptionsModel =
-        new CreateCollectionOptions.Builder().name("testString").description("testString").build();
```
Comment (apaparazzi0329) [inline visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptionsTest.java:54]: This file should have no changes

File: visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/UpdateCollectionOptionsTest.java [hunk 18]
```diff
@@ -55,11 +55,12 @@ public void testUpdateCollectionOptions() throws Throwable {
             .collectionId("testString")
             .name("testString")
             .description("testString")
+            .trainingStatus(trainingStatusModel)
```
Comment (apaparazzi0329) [inline visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/UpdateCollectionOptionsTest.java:58]: Use of TrainingStatus and its assertEquals should be removed

**Review by apaparazzi0329 — CHANGES_REQUESTED** [review]

File: visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptionsTest.java (hunk 17 above)
Comment (kevinkowa) [reply visual-recognition/src/test/java/com/ibm/watson/visual_recognition/v4/model/CreateCollectionOptionsTest.java:54]: Getting rid of trainingStatus there, and the Java-format script fixed its format too

File: speech-to-text/src/main/java/com/ibm/watson/speech_to_text/v1/model/RecognizeOptions.java [hunk 19]
```diff
@@ -101,17 +125,16 @@
...
+    /** pt-BR_Telephony. */
+    String PT_BR_TELEPHONY = "pt-BR_Telephony";
     /** zh-CN_BroadbandModel. */
     String ZH_CN_BROADBANDMODEL = "zh-CN_BroadbandModel";
     /** zh-CN_NarrowbandModel. */
     String ZH_CN_NARROWBANDMODEL = "zh-CN_NarrowbandModel";
   }
 
   protected InputStream audio;
-
```
Comment (apaparazzi0329) [inline speech-to-text/src/main/java/com/ibm/watson/speech_to_text/v1/model/RecognizeOptions.java:111]: Make sure this is not a hand edit being deleted. Why was this serialized before?

**Review by apaparazzi0329 — CHANGES_REQUESTED** [review]
The only changes I have left to review is Assistant but I will submit the changes I have requested so far

**Review by repjarms — COMMENTED** [review]
Overall the PR looks good. I asked a few questions within the changes that could use some clarification before approval.

File: assistant/src/main/java/com/ibm/watson/assistant/v2/Assistant.java [hunk 20]
```diff
@@ -55,7 +55,7 @@
  */
 public class Assistant extends BaseService {
 
-  public static final String DEFAULT_SERVICE_NAME = "assistant";
+  public static final String DEFAULT_SERVICE_NAME = "conversation";
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/main/java/com/ibm/watson/assistant/v2/Assistant.java:58]: good catch

File: assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextSkillTest.java [hunk 21]
```diff
@@ -43,7 +43,7 @@ public void testMessageContextSkill() throws Throwable {
             .userDefined(
                 new java.util.HashMap<String, Object>() {
                   {
-                    put("foo", "testString");
+                    put("foo", TestUtilities.createMockMap());
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextSkillTest.java:46]: the unit tests are generated

File: natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java [hunk 22]
```diff
@@ -121,7 +139,7 @@ public Builder keywords(KeywordsOptions keywords) {
      * @param metadata the metadata
      * @return the Features builder
      */
-    public Builder metadata(Map<String, Object> metadata) {
+    public Builder metadata(HashMap<String, Object> metadata) {
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java:142]: hmm good question, with a map you have multiple choices such as LinkedHashMap, TreeMap, and HashMap

File: assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextStatelessTest.java [hunk 23]
```diff
@@ -65,7 +65,7 @@ public void testMessageContextStateless() throws Throwable {
             .userDefined(
                 new java.util.HashMap<String, Object>() {
                   {
-                    put("foo", "testString");
+                    put("foo", TestUtilities.createMockMap());
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextStatelessTest.java:68]: this got generated

File: assistant/src/test/java/com/ibm/watson/assistant/v1/AssistantTest.java [hunk 24]
```diff
@@ -3769,175 +3815,6 @@ public void testListAllLogsNoOptions() throws Throwable {
     assistantService.listAllLogs(null).execute();
   }
 
-  @Test
-  public void testUpdateDialogNodeNullableWOptions() throws Throwable {
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v1/AssistantTest.java:3773]: Ah this is a manual unit test I added for the UpdateDialogNode allowing nulls

File: natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java (hunk 22 above)
[the remark this answers is not in the document]
Comment (apaparazzi0329) [reply natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java:142]: In that case this would be a breaking change for users who use LinkedHashMap or TreeMap. Would their be any issues with reverting it back?

[the remark this answers is not in the document]
Comment (kevinkowa) [reply natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java:142]: nothing would break if we change it back, this might need to be a hand edit to keep in mind

File: assistant/src/test/java/com/ibm/watson/assistant/v1/model/UpdateDialogNodeTest.java [hunk 25]
```diff
@@ -1,366 +0,0 @@
-/*
```
Comment (apaparazzi0329) [inline assistant/src/test/java/com/ibm/watson/assistant/v1/model/UpdateDialogNodeTest.java:1]: Why was this file deleted?

**Review by apaparazzi0329 — DISMISSED** [review]
Its looking good to me!

File: assistant/src/test/java/com/ibm/watson/assistant/v1/model/UpdateDialogNodeTest.java (hunk 25 above)
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v1/model/UpdateDialogNodeTest.java:1]: Good catch, it's part of a manual test added for the update dialog node nullable feature

File: assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextSkillTest.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextSkillTest.java:46]: It's also an object in the docs https://cloud.ibm.com/apidocs/assistant/assistant-v2#messagestateless.

File: assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextStatelessTest.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (kevinkowa) [reply assistant/src/test/java/com/ibm/watson/assistant/v2/model/MessageContextStatelessTest.java:68]: It's also an object in the docs https://cloud.ibm.com/apidocs/assistant/assistant-v2#messagestateless.

File: natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java (hunk 22 above)
[the remark this answers is not in the document]
Comment (kevinkowa) [reply natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java:142]: I created an issue to investigate this problem https://github.ibm.com/Watson/developer-experience/issues/9416.
In the mean time I'm going to apply a hand edit change

[the remark this answers is not in the document]
Comment (kevinkowa) [reply natural-language-understanding/src/main/java/com/ibm/watson/natural_language_understanding/v1/model/Features.java:142]: ah I remember now, it's a hand edit. The generator uses `protected MetadataOptions metadata;` but we change it to a `Map`. Although, here I guess it was a mistake. Fixing it!

**Comment by adriancampos1** [comment]
Can you all share if this May release is now available or an ETA? Thank you in advance.

**Review by apaparazzi0329 — APPROVED** [review]
Changes look good

**Review by repjarms — APPROVED** [review]
LGTM

**Comment by apaparazzi0329** [comment]
@adriancampos1 ETA is end of day, barring any CI issues that may present themselves after merging to master

**Comment by kevinkowa** [comment]
I'm running local IT to make sure everything is good to go. After that I'll merge and it will take some time to be up in maven
